src/battle_ai_main.c

```diff
@@ -1372,7 +1372,7 @@ static s16 AI_CheckBadMove(u8 battlerAtk, u8 battlerDef, u16 move, s16 score)
         case EFFECT_AURORA_VEIL:
             if (gSideStatuses[GetBattlerSide(battlerAtk)] & SIDE_STATUS_AURORA_VEIL
               || PartnerHasSameMoveEffectWithoutTarget(BATTLE_PARTNER(battlerAtk), move, AI_DATA->partnerMove)
-              || !(gBattleWeather & B_WEATHER_HAIL))
+              || !(gBattleWeather & B_WEATHER_HAIL && gBattleWeather & B_WEATHER_SNOW))
```

Comment:
It looks like the AI will always subtract 10 score unless both snow and hail are set up here.

Reply:
when both are **NOT** set up.

Reply:
If gBattleWeather & B_WEATHER_HAIL is true and gBattleWeather & B_WEATHER_SNOW is false, then the entire thing will be false. Due to the !, it'll pass and the score will be subtracted by 10 despite one of the weathers being up. This doesn't seem intentional.

Reply:
yeah youre right I think. I did feel like this was wrong when I wrote it but I was to stupid to resolve it in my head 😄